### PR TITLE
ENH: Bump to py39, add conda config

### PR DESCRIPTION
**Pull request by mgxd**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -72,66 +73,66 @@ ENV PATH="${CONDA_PATH}/bin:$PATH" \
...
-                     psutil=5.8 \
+                     packaging=21.3 \
+                     pandas=1.4 \
+                     pandoc=2.18 \
+                     pbr=5.9 \
+                     pip=22.0 \
+                     pockets=0.9 \
+                     psutil=5.9 \
                      pydot=1.4 \
                      pydotplus=2.0 \
```
Comment (effigies) [inline Dockerfile]: I think we can drop `pydotplus` at this point.

**Comment by effigies** [comment]
@mgxd Since we need to test, why not just make a `py39` branch and start tagging it with `py39_1.0.0`?

**Comment by mgxd** [comment]
sure, just pushed [`py39_4.12.0rc0`](https://github.com/nipreps-containers/miniconda/releases/tag/py39_4.12.0rc0) if you want to give it a try

**Comment by oesteban** [comment]
May I suggest to install conda, right after [mamba](https://github.com/mamba-org/mamba), and then use mamba for the long list of dependencies.

Mamba should resolve the dependencies way faster than conda.

**Comment by effigies** [comment]
Are we worried about runtime for this build?

**Comment by oesteban** [comment]
> Are we worried about runtime for this build?

Not particularly, but it seems a good idea to speed it up and have mamba available to downstream images.

File: Dockerfile [hunk 2]
```diff
@@ -72,66 +73,65 @@ ENV PATH="${CONDA_PATH}/bin:$PATH" \
...
-                     sphinx=4.2 \
-                     sphinx_rtd_theme=1.0 \
-                     "svgutils>=0.3.4,<0.4" \
-                     toml=0.10 \
-                     traits=6.2 \
-                     zlib=1.2 \
-                     zstd=1.5; sync && \
+COPY condarc /root/.condarc
+
+RUN ${CONDA_PATH}/bin/conda install \
```
Comment (mgxd) [inline Dockerfile]: I tested yesterday with the following:
```suggestion
RUN ${CONDA_PATH}/bin/conda install mamba -n base \
+    && mamba install -y \
```

and did not really see much if any speedup

File: Dockerfile [hunk 3]
```diff
@@ -72,66 +73,65 @@ ENV PATH="${CONDA_PATH}/bin:$PATH" \
...
+                    libxslt=1.1 \
+                    lockfile=0.12 \
+                    matplotlib=3.5 \
+                    mkl=2022.1 \
+                    mkl-service=2.4 \
+                    nibabel=3.2 \
+                    nilearn=0.9 \
+                    nipype=1.8 \
+                    nitime=0.9 \
+                    nodejs=17.9 \
```
Comment (mgxd) [inline Dockerfile]:
```suggestion
                    nodejs=16 \
```

**Comment by effigies** [comment]
@mgxd I think this is basically good to go, but I would suggest we not tie ourselves to miniconda version strings. What do we do when we want to update our image without bumping the miniconda version? I think we could just do `py39_1.0.0` and have a version history per-Python-minor.

File: Dockerfile (hunk 2 above)
Comment (mgxd) [reply Dockerfile]: I'll add in mamba just to have the option to help with package resolution in subsequent images

**Comment by mgxd** [comment]
@effigies WDYT about timestamping the version, e.g. `py39_05252022`. I feel like it gives a quick ballpark to how updated the packages are

**Comment by effigies** [comment]
Sure, but let's calver it so it sorts. `YYMM.0`?

**Comment by oesteban** [comment]
I think we will need to clean up the docker layer caching action :S

**Comment by mgxd** [comment]
I pushed a release, @oesteban merge whenever

**Pull request merged**